app/javascript/components/firmware-registry-add-edit.jsx

```diff
@@ -0,0 +1,135 @@
+      type: 'FormButtons.saveable',
+      payload: formState.valid,
+    });
+    props.dispatch({
+      type: 'FormButtons.pristine',
+      payload: formState.pristine,
+    });
+    props.dispatch({
+      type: 'FormButtons.callbacks',
+      payload: { addClicked: () => submitValues(formState.values) },
+    });
+  };
+
+  return (
+    <Grid fluid>
+      <Row>
+        <MiqFormRenderer
+          initialValues={initialValues}
+          schema={
+              {
+                fields: [
+                  {
+                    component: componentTypes.SELECT_COMPONENT,
+                    name: 'type',
+                    options: [{ value: 'FirmwareRegistry::RestApiDepot', label: __('Rest API Depot') }],
+                    label: __('Registry Type'),
+                    placeholder: __('Nothing selected'),
+                    isSearchable: true,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                    validateOnMount: true,
+                    isDisabled: true,
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'name',
+                    label: __('Name'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'url',
+                    label: __('URL'),
+                    placeholder: 'https://example.com:1234/images',
+                    maxLength: 2048,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'userid',
+                    label: __('Username'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                  },
+                  {
+                    component: componentTypes.TEXT_FIELD,
+                    name: 'password',
+                    label: __('Password'),
+                    maxLength: 128,
+                    validate: [{
+                      type: validatorTypes.REQUIRED,
+                      message: __('Required'),
+                    }],
+                    type: 'password',
+                  },
+                ],
+              }
+          }
+          onSubmit={() => {}}
+          onCancel={() => {}}
+          buttonsLabels={{
+            submitLabel: __('Create'),
+          }}
+          className=""
```

Comment:
empty classname

Reply:
This actually needs to be here in order to override miq form renderer default class name. (To make it 100% width of the modal instead of 50%)

Reply:
OK :)

(I don't think *not* providing a className should ever set more classes then providing one (maybe call it `overrideClassName`?), but that's probably not relevant *here* :).)

Reply:
I do agree that its unfortunate. Passing empty class name will actually remove the default one. The default case does not consider using it within a modal. (https://github.com/ManageIQ/manageiq-ui-classic/pull/5650)